pubber/src/main/java/daq/pubber/ManagerBase.java

```diff
@@ -80,10 +80,16 @@ public void error(String message) {
   protected void updateInterval(Integer sampleRateSec) {
     int reportInterval = ofNullable(sampleRateSec).orElse(DEFAULT_REPORT_SEC);
     int intervalSec = ofNullable(options.fixedSampleRate).orElse(reportInterval);
+    if (intervalSec < DISABLED_INTERVAL) {
+      error(format("Dropping update interval, negative sample rate %ds", intervalSec));
+      return;
+    }
     if (periodicSender == null || intervalSec != sendRateSec.get()) {
       cancelPeriodicSend();
       sendRateSec.set(intervalSec);
-      startPeriodicSend();
+      if (intervalSec > DISABLED_INTERVAL) {
```

Comment:
This check seems redundant because of the previous check. Also, the case where intervalSec == DISABLED_INTERVAL likely isn't handled correctly.

Reply:
It looks ok to me. Not start periodic senders and only cancel when intervalSec == DISABLED_INTERVAL

Reply:
Ah, got it.